### PR TITLE
add spring dependency injection example

### LINKED ISSUE #3: Spring Dependency Injection
Napravi spring aplikaciju koja :
   1. ima dve POJO klase Airplane i Pilot
        1. Airplane klasa ima sledece atribute
            - svog pilota
            - broj putnika
        2. Pilot klasa ima sledece atribute
            - ime pilota
   2.  preko xml konfiguracije definise dva beana airplane i pilot, gde se povezuju Airplane i Pilot bean
        - u konfiguraciji takodje podesi ime Pilota (npr. Anja ) i broj putnika u avionu
   3. u main metodi preko context-a cita Airplane bean, iz tog beana cita pilota i ispisuje ime pilota

### PR DESCRIPTION
**Pull request by anjaradenovic**
closes #3

### CODE REVIEW

File: src/main/resources/Beans.xml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<?xml version = "1.0" encoding = "UTF-8"?>
+
+<beans xmlns = "http://www.springframework.org/schema/beans"
+       xmlns:xsi = "http://www.w3.org/2001/XMLSchema-instance"
+       xsi:schemaLocation = "http://www.springframework.org/schema/beans
+   http://www.springframework.org/schema/beans/spring-beans-3.0.xsd">
+
+    <bean id = "airplane1" class = "com.endava.exercise.Airplane">
```
Comment (oskarv) [inline src/main/resources/Beans.xml]: having "1" in id value is anti pattern.
What do you think, why?

File: src/main/resources/Beans.xml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+        <property name = "pilot" value = "#{pilot1}"/>
+        <property name = "numberOfPassengers" value = "25"/>
+    </bean>
+
+    <bean id = "pilot1" class = "com.endava.exercise.Pilot">
```
Comment (oskarv) [inline src/main/resources/Beans.xml]: Same here

**Review by oskarv — CHANGES_REQUESTED** [review]

File: src/main/resources/Beans.xml (hunk 1 above)
Comment (anjaradenovic) [reply src/main/resources/Beans.xml]: I must admit that it does look wrong :D and I google it, but I couldn't find "officially" why. I suppose that it is ambiguous and non-descriptive - later when using the beans, what is the difference between "airplane1" and some other "airplane2"?

I have renamed it to be more descriptive.

Comment (oskarv) [reply src/main/resources/Beans.xml]: "...1" in the name only make sense if there are "....2" , "....3" and so on, but that's not the right way to get multiple airplane instances.
Ability of a component to be unique / singular or to be non-unique / plural is handled with bean scopes (Singleton and Prototype) , it shouldn't be handled by bean names / id's ("airplane1", "airplane2" etc ). 
You could have left this bean to be id="airplane" but made it a Prototype Scope which ensures that you can create many different instances of airplanes.

What you did to fix this is acceptable ( but is somewhat a work-around ) , you used specific plane type, in which case it's fine to have a Singleton.

**Review by oskarv — APPROVED** [review]
please merge this PR

**Pull request merged**